src/datasets/utils/file_utils.py

```diff
@@ -1103,7 +1103,12 @@ def read_with_retries(*args, **kwargs):
             try:
                 out = read(*args, **kwargs)
                 break
-            except (ClientError, TimeoutError) as err:
+            except (
+                ClientError,
+                TimeoutError,
```

Comment:
(nit) Those two correspond to `aiohttp.client_exceptions.ClientError` and `asyncio.TimeoutError`. I think it's best to explicitly name them as you did with `requests.exceptions.ConnectTimeout`. It was not obvious for me at first (I wondered what was the difference between those 4 only to figure out it was not only `requests` errors)